Make Library great again

pva701 opened the pull request:
## Description

<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

https://issues.serokell.io/issue/EDNA-31

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

pva701 committed:
Initialise context actions

Problem: we don't have context actions
for different elements (for now it's only rows in table).

Solution: add basic context actions machinery.

pva701 committed:
Change colors according to mockup. Change Header height.

pva701 committed:
Implement new layout for Library/Methodologies

Problem: current layout of Methodologoies doesn't
match figma mockup.

Solution: implement tables instead of plates.

pva701 committed:
Edit dialog for methodology and project

Problem: we don't have dialogs.

Solution: add dialog layout.

pva701 committed:
Refactor dialogs. Methodology description dialog

Problem: dialogs have to be fit in the same schema
and machinery.

Solution: abstract out some parts.

pva701 committed:
Delete methodology dialog

Problem: we don't have Delete dialog.

Solution: add one more dialog type.

pva701 committed:
Add methodology link addition/editing dialog

Problem: it's not possible to edit links.

Solution: add dialog for it.

pva701 committed:
Compounds table layout

Problem: compounds table doesn't match figma design.

Solution: adjust it.

pva701 committed:
Empty placeholder for library tables

Problem: we don't have placeholders for no data.

Solution: add them.

gromakovsky commented:
@pva701 can you please also update all timestamps in the tables to show time on hover (like GitHub does for example)? We need it to test that we handle timezones properly.

sashasashasasha151 commented on frontend/src/components/ContextActions/ContextActions.tsx:
```diff
@@ -0,0 +1,18 @@
+
+export function ContextActions({ actions }: ContextActionsProps): React.ReactElement {
+  return (
+    <div className="contextActions__wrapper">
```
As I undestand `contextActions` is enough here. Like it is a block, not element

sashasashasasha151 commented on frontend/src/components/ContextActions/ContextActions.scss:
```diff
@@ -0,0 +1,63 @@
+  &__btn {
+    padding: 0 (22 * $px);
+    font-family: inherit;
+    font-size: 100%;
```
What for? (I know that you will say that it is not minimum code contest, but here you even have no text)

sashasashasasha151 commented on frontend/src/components/ContextActions/ContextActions.scss:
```diff
@@ -0,0 +1,64 @@
+  &__content {
+    position: absolute;
+    left: 50%;
+    z-index: 2;
```
Make nothing as I can see

sashasashasasha151 commented on frontend/src/components/ContextActions/ContextActions.scss:
```diff
@@ -0,0 +1,64 @@
+    box-shadow: 0 0 (5 * $px) rgba(151, 151, 151, 0.3);
+  }
+
+  &__btn:focus ~ &__content,
+  &__btn ~ &__content:active {
+    display: flex;
+  }
```
If after opened action menu I click on button again, I can't close it

pva701 replied:
:loh:

sashasashasasha151 commented on frontend/src/components/ContextActions/ContextActions.scss:
```diff
@@ -0,0 +1,63 @@
+      margin-right: 14 * $px;
+    }
+
+    &:first-child {
+      border-top-left-radius: 4 * $px;
+      border-top-right-radius: 4 * $px;
+    }
+
+    &:last-child {
+      border-bottom-right-radius: 4 * $px;
+      border-bottom-left-radius: 4 * $px;
+    }
```
Maybe I don't understand smth but looks like it do nothing

pva701 replied:
I like you!

pva701 replied:
How do you realise it ? it does nothing indeed. it was remnants from attempts to make items hoverable (where hover has rounded corners)

sashasashasasha151 commented on frontend/src/components/Table/Table.scss:
```diff
@@ -37,48 +37,31 @@
+    max-height: 50 * $px;
     padding: (16 * $px) 0 (16 * $px) $cell-indent;
+    font-size: 14 * $px;
+    font-weight: 400;
+    line-height: 18 * $px;
```
Looks like not needed

pva701 replied:
I like you!

pva701 replied:
yes and no, now it seems that 18px follows from 16*2 paddings, but I had some troubles in this place, so I want it to be explicitly specified in case if we change paddings and the row height.

sashasashasasha151 commented on frontend/src/components/Table/Table.tsx:
```diff
@@ -47,23 +55,22 @@ export function Table<T extends object>({
-              className={cx({ ednaTable__row_withHover: mode === "bordered" })}
+              className={`ednaTable__row ${
+                i % 2 === 0 ? "ednaTable__row_odd" : "ednaTable__row_even"
+              }`}
```
`:nth-child(even)`

sashasashasasha151 commented on frontend/src/components/Table/Table.tsx:
```diff
@@ -47,23 +55,22 @@ export function Table<T extends object>({
             >
               {row.cells.map(cell => {
-                return (
+                const manualCell =
+                  cell.column.id &&
+                  columnExtras &&
+                  columnExtras[cell.column.id] &&
+                  columnExtras[cell.column.id].manualCellRendering;
+                return manualCell ? (
+                  cell.render("Cell", cell.getCellProps())
+                ) : (
```
Don't what to think what is going on)))

pva701 replied:
Calm, we are developing Edna project. Everything is under control.

sashasashasasha151 commented on frontend/src/pages/library/LibraryPage.tsx:
```diff
@@ -1,23 +1,29 @@
 import React, { FunctionComponent, useState } from "react";
-import { useRecoilValue } from "recoil";
+import { useRecoilValue, useSetRecoilState } from "recoil";
```
Library page has 400+ lines, lets split it somehow

pva701 replied:
horizontally, vertically, or diagonally? :putya:

sashasashasasha151 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/25207759/113486587-4e19ac80-94bc-11eb-8712-0280a7eb1cc1.png)
Edit out of the screen

pva701 committed:
Stylistic improvements

Problem: there are several issues wuth css, and ContextActions interaction.

Solution: fix them.
